### ISSUE TITLE
v1.6.0  - RangeError: Invalid array length

### ISSUE THREAD

**Issue by MikeNats** [strongloop/strong-soap#149]
In the new release v1.6.0 
 ` this.includes = this.includes.concat(source.includes);`  was introduced at
`.../node_modules/strong-soap/src/parser/xsd/schema.js:33`

When I `node  .` my loopback server  throws an error with the below stack trace: 

```
.../node_modules/strong-soap/src/parser/xsd/schema.js:33
      this.includes = this.includes.concat(source.includes);

RangeError: Invalid array length
    at Schema.merge (...node_modules/strong-soap/src/parser/xsd/schema.js:33:37)
    at ...node_modules/strong-soap/src/parser/wsdl.js:192:44
    at baseMergeDeep (.../node_modules/lodash/lodash.js:3655:11)
    at .../node_modules/lodash/lodash.js:3615:11
    at .../node_modules/lodash/lodash.js:4917:15
    at baseMerge (.../node_modules/lodash/lodash.js:3612:7)
    at baseMergeDeep (.../node_modules/lodash/lodash.js:3701:9)
    at .../node_modules/lodash/lodash.js:3615:11
    at .../node_modules/lodash/lodash.js:4917:15
    at baseMerge (.../node_modules/lodash/lodash.js:3612:7)
    at .../node_modules/lodash/lodash.js:13464:7
    at Function.<anonymous> (.../node_modules/lodash/lodash.js:4865:13)
    at apply (.../node_modules/lodash/lodash.js:467:27)
    at Function.mergeWith (.../node_modules/lodash/lodash.js:6575:16)
    at WSDL.<anonymous> (.../node_modules/strong-soap/src/parser/wsdl.js:191:11)
    at Function.load (.../node_modules/strong-soap/src/parser/wsdl.js:361:23)
```

It works fine with v1.5.0.
Any idea?
I use `...`  to hide obvious information

**Comment by raymondfeng** [comment]
@MikeNats Can you try to see if 1.6.1 fixes your problem?

Your wsdl/xsd files will be helpful to reproduce the problem.

**Comment by MikeNats** [comment]
`npm ERR! notarget No compatible version found: strong-soap@1.6.1`

**Comment by raymondfeng** [comment]
It may take some time for npm to see it.

**Comment by MikeNats** [comment]
`RangeError: Invalid array length`  error   seems to occur because of a recursive import of files.
and throws as the array is growing exponentially.

As you use lodash  maybe a quick fix is to use  `_.uniqeBy`  with second argument  `namespace`.
so something like the below just to make sure that you want concat an array that yits size is >2^32.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_array_length 

`this.includes = _.uniqueBy(this.includes.concat(source.includes), 'namespace');`

**Comment by raymondfeng** [comment]
@MikeNats Can you provide your WSDL/XSD files so that I can reproduce the problem and fix it accordingly instead of guessing and trying?

**Comment by snagi** [comment]
@raymondfeng, Please see the WSDL/XSD and associated datasource.json file for configuring the connector in attached zip file.
[WSDL-and-datasource.zip](https://github.com/strongloop/strong-soap/files/1830384/WSDL-and-datasource.zip)

**Comment by raymondfeng** [comment]
@MikeNats @snagi Please verify https://github.com/strongloop/strong-soap/pull/150. 

I have included your test wsdl/xsd files in the PR. Please note that I had to fix some dangling references.

**Comment by raymondfeng** [comment]
Fixed by #150
